src/systems/exit-on-blur.js

```diff
@@ -2,18 +2,26 @@ AFRAME.registerSystem("exit-on-blur", {
   init() {
     this.onBlur = this.onBlur.bind(this);
     this.onFocus = this.onFocus.bind(this);
+    this.onTimeout = this.onTimeout.bind(this);
+
+    this.isOculusBrowser = navigator.userAgent.match(/Oculus/);
 
     window.addEventListener("blur", this.onBlur);
     window.addEventListener("focus", this.onFocus);
 
     this.exitTimeout = null;
   },
 
+  tick() {
+    if (this.isOculusBrowser) {
+      clearTimeout(this.exitTimeout);
+      this.exitTimeout = setTimeout(this.onTimeout, 15 * 1000);
+    }
+  },
+
   onBlur() {
     if (this.el.isMobile) {
-      this.exitTimeout = setTimeout(() => {
-        this.el.dispatchEvent(new CustomEvent("exit"));
-      }, 30 * 1000);
+      this.exitTimeout = setTimeout(this.onTimeout, 30 * 1000);
```

Comment:
Should probably clearTimeout here so you dont end up accidentaly firing it multiple times.

Reply:
Yeah, good point.